eng/pipelines/templates/steps/cache-maven-repository.yml

```diff
@@ -1,15 +1,6 @@
 steps:
-- script: |
-    echo "##vso[task.setvariable variable=Maven.RepositoryPath;]%USERPROFILE%\.m2\repository"
-  condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'))
-  displayName: Detecting Maven repository on Windows
-- script: |
-    echo "##vso[task.setvariable variable=Maven.RepositoryPath;]$HOME/.m2/repository"
-  condition: and(succeeded(), ne(variables['Agent.OS'], 'Windows_NT'))
-  displayName: Detecting Maven repository on Linux and macOS
-- task: CacheBeta@0
+- task: Cache@2
   inputs:
-    key: $(Agent.JobName)|$(CacheSalt)|$(Build.SourcesDirectory)/sdk/**/pom.xml|$(Build.SourcesDirectory)/pom*.xml
-    path: $(Maven.RepositoryPath)
-  displayName: 'Download/upload cache'
-  condition: and(succeeded(), ne(variables['EnableCaching'], 'false'))
\ No newline at end of file
+    key: 'maven | $(Agent.OS) | $(Build.SourcesDirectory)/eng/versioning/external_dependencies.txt'
```

Comment:
@mitchdenny I know you mentioned keeping a salt but I don't think that is necessary as long as you have some file like the external dependencies you can update to invalidate the cache. However I'm considering adding in the Agent.JobName instill because for steps like Build vs Test vs Analyze the sometimes have a different set of packages that is interesting.  What is your thought? Do you think it is necessary?

Reply:
I added the CacheSalt variable so we can disable/invalidate the cache if needed. We can keep watch of the caching to see if some of the jobs can potentially gain more by having their own cache instead of sharing the case of other jobs. If that is the case we can add in the JobName back into the key.